### PR TITLE
Add SelectMenu.LoadingAnimation

### PR DESCRIPTION
**Pull request by emplums**
This PR adds a `SelectMenu.LoadingAnimation` component

### Screenshots
image

### Notes
I thought about having the loading animation baked into the `SelectMenu.Modal` and just show and hide it if a `loading` prop was set to true. 
```
const SelectMenuModal = ({children, theme, width, loading, ...rest}) => {
  return (
    <ModalWrapper theme={theme} {...rest} role="menu">
      <Modal theme={theme} width={width}>
        {loading ? <LoadingAnimation/> : children}
      </Modal>
    </ModalWrapper>
  )
}

```
When going down this path, I realized I needed to decide whether to conditionally show the loading animation or the `{children}` content, or show both of them. Conditionally showing the loading animation or the `children` content wouldn't work because our SelectMenu.Modal API expects users to pass in the `SelectMenu.Filter` as a child, and on .com we tend to show the filter while the content is loading (I'm assuming this is because we want to be able to put focus on the filter input as soon as the menu is shown for a11y reasons).

On the other hand, if we optimistically place the loading animation after `{children}` we run the risk of something else being in the modal content (like a footer) that would appear above the loading animation. In this scenario, we could ask users of the component to make sure not to show unnecessary UI in the modal during the loading state and hope that gets handled properly in the consuming application.

I think the best approach is to do neither, and just make the `LoadingAnimation` component available on the public component API. Users can include the component and decide how they want to handle the content of their modals during the loading state. This opens up some risk of the consuming application putting the loading animation in the wrong place, but I think it's a lot of straightforward of a component API and feels in line with the style of the rest of the SelectMenu API.

### Merge checklist
- [ ] Added or updated TypeScript definitions (`index.d.ts`) if necessary
- [ ] Added/updated tests
- [ ] Added/updated documentation
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/components/blob/main/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

File: docs/content/SelectMenu.md [hunk 1]
```diff
@@ -348,3 +348,34 @@ SelectMenu.Header components get `COMMON` system props. Read our [System Props](
...
+
+**Note**:  You will need to handle showing/hiding the appropriate modal content for your application during the loading state. We recommend always showing the `SelectMenu.Filter` and `SelectMenu.Header`  (if used) and hiding the rest of the modal content during the loading state.
+
+```jsx live
+<SelectMenu>
+  <Button as="summary">Projects</Button>
+  <SelectMenu.Modal>
+    <SelectMenu.Header>Projects</SelectMenu.Header>
+    <SelectMenu.Filter placeholder="Filter projects" value="" aria-label="Filter Projects"/>
+    {true ? <SelectMenu.LoadingAnimation/> :
```
Comment (emplums) [inline docs/content/SelectMenu.md:363]: @BinaryMuse is there a way to set a variable in the code example here like `const loading = true` ? I tried adding it above line 358 but still got a `loading is undefined` error 🤔 I thought there was a way to do this but maybe not. If there is, I'd like to take everything from line 364 to line 370 and put it in a const so I can do this:

`{loading ? <SelectMenu.LoadingAnimation/> : modalContent}`
to make it look a little cleaner

Comment (BinaryMuse) [reply docs/content/SelectMenu.md:363]: Unfortunately not; without [the `noinline` annotation](https://primer.style/doctocat/usage/live-code#no-inline) Doctocat treats the whole code block as a single expression.

**Review by BinaryMuse — APPROVED** [review]
Looks good! I think your train of thought on showing the indicator is solid. When and how to manage loading states is another good candidate for "Primer Patterns" :)

File: docs/content/SelectMenu.md (hunk 1 above)
Comment (BinaryMuse) [reply docs/content/SelectMenu.md:363]: (But you could rewrite the example using `noinline`)

**Pull request merged**